explorer/templates_test.go

```diff
@@ -0,0 +1,59 @@
+package explorer
+
+import (
+	"testing"
+)
+
+func TestamountAsDecimalPartsTrimmed(t *testing.T) {
+	test1 := amountAsDecimalPartsTrimmed(314159000, 2, false)
+	test2 := amountAsDecimalPartsTrimmed(76543210000, 2, false)
+	test3 := amountAsDecimalPartsTrimmed(76643210000, 2, false)
+	test4 := amountAsDecimalPartsTrimmed(654321000, 1, false)
+	test5 := amountAsDecimalPartsTrimmed(987654321, 8, false)
+	test6 := amountAsDecimalPartsTrimmed(987654321, 2, false)
+	test7 := amountAsDecimalPartsTrimmed(90765432100, 2, false)
+	test8 := amountAsDecimalPartsTrimmed(9076543200000, 2, false)
+	test9 := amountAsDecimalPartsTrimmed(907654320, 7, false)
+	test10 := amountAsDecimalPartsTrimmed(1234590700, 2, false)
+
+	if test1[0]+"."+test1[1] != "3.14" {
+		t.Error("Wrong output, got" + test1[0] + "." + test1[1])
```

Comment:
The error message should state expected and actual.  e.g. "amountAsDecimalPartsTrimmed failed for %d (%d decimals).  Got %s.%s, expected %s.%s.".